File: practicas/practica5-poo/poo.js
class Perro {
    constructor(nombre, color, edad){
        this.nombre = nombre;
        this.color = color;
        this.edad = edad;
    }

    ladrar(saludo){
        return `${this.nombre}: ${saludo}`
    }

    toString(){
        return `Hola, mi nombre es ${this.nombre} y yo soy de color ${this.color}`
    }
}

class Husky extends Perro{
    jalarTrineo(){
        return `${this.nombre} esta jalando el trineo` 
    }

    ladrar(){
        return `Nuevo ladrido: arf`
    }
}

class Bulldog extends Perro{
    morder(){
        return `${this.nombre} esta mordiendo!` 
    }
}

let perro = new Perro();
perro.ladrar();
perro.ladrar('guau!')

File: practicas/practica5-poo/poo.py
# perro = ['Balu', 'Blanco', 12]
# perro = {
#     'nombre': 'Balu',
#     'edad': 12
# }

# perro['nombre']

class Perro:
    def __init__(self, nombre, edad, color) -> None:
        self.nombre = nombre
        self.edad = edad
        self.color = color
    
    def ladrar(self, ladrido):
        return f'{ladrido} de color {self.color}!!'

    def __str__(self):
        return f'Hola! me llamo {self.nombre}'

class Husky(Perro):
    def jalar_trineo(self):
        return f'{self.nombre} esta jalando el trineo!'

class Bulldog(Perro):
    def morder(self):
        return f'{self.nombre} esta mordiendo.'

     